### PR TITLE
Updating device api

### PR DESCRIPTION
**Pull request by plauche**
- Updating and cleaning up device api guide
- Moving details about how to document and how to test into their own docs

### CODE REVIEW

**Comment by plauche** [comment]
@CatGarab @jacoffey3 This PR is still a WIP, but I'd like some feedback on the splitting out of the testing and documentation sections. These sections (or the bulk of them) seemed like they belong in separate guides which can be referenced elsewhere, and contributing seemed like the natural place for them. Thoughts?

**Comment by CatGarab** [comment]
I like it :+1:

**Comment by jacoffey3** [comment]
Seconded 👍

File: docs/contributing/documentation.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+Documentation
+=============
+
+TODO: Fill out the rest of this
```
Comment (plauche) [inline docs/contributing/documentation.rst]: Intentionally leaving this incomplete, will be addressed in https://trello.com/c/TEP31d85/920-add-a-doc-for-contributing-to-the-docs.

File: docs/contributing/testing.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+Testing
+=======
+
+TODO: Fill out the rest of this
```
Comment (plauche) [inline docs/contributing/testing.rst]: Intentionally leaving incomplete, will be addressed in https://trello.com/c/WGCoXKM0/929-add-doc-for-code-testing-practices.

**Comment by plauche** [comment]
[Review/Update/Remove device API guide](https://trello.com/c/E9strML9/876-review-update-remove-device-api-guide)

File: docs/deep-dive/apis/device-api-guide.rst [hunk 3]
```diff
@@ -130,147 +76,136 @@ While each API is highly device-specific, our goal is to keep the overall stylin
...
-Within each test set should be three files:
+Documentation
+-------------
 
-    - ``<test-set>.c`` - The file containing the actual tests
-    - ``sysfs.c`` - Stub functions for the underlying `sysfs` calls
-    - ``stubs.cmake`` - Makes the stub functions available to the test builder/runner
+Documentation for new device APIs should be added to the `docs/deep-dive/apis/device-api` folder.
 
+Each API will at least one doc:
```
Comment (CatGarab) [inline docs/deep-dive/apis/device-api-guide.rst]: nit: missing a word

File: docs/deep-dive/apis/device-api-guide.rst [hunk 4]
```diff
@@ -130,147 +76,136 @@ While each API is highly device-specific, our goal is to keep the overall stylin
...
+    - ``<new-api>.rst`` - API's users guide
 
-To run the tests the same way that CircleCI does, navigate to the top level of the Kubos repo and issue this command::
+APIs which are written in C will require a second doc:
 
-    $ python $PWD/tools/build.py --all-tests
-    
-Rust
-^^^^
+    - ``<new-api>_api.rst`` - Source level API doc
```
Comment (CatGarab) [inline docs/deep-dive/apis/device-api-guide.rst]: nit: hyphenate "Source-level"

File: docs/deep-dive/apis/device-api-guide.rst [hunk 5]
```diff
@@ -101,27 +57,17 @@ New APIs should be located in a new folder within the `apis` folder of the `Kubo
...
+    |   +-- <new-api>\
     |
     +-- cargo-kubos\
     +-- ccan\
     +-- cmocka\
-    
-See the :doc:`C module development doc <../sdk-docs/sdk-c>` for steps to create a new Kubos module in C.
 
-APIs written in Rust will reside under the same parent folder, but their files will be generated with ``cargo``.
+APIs may be written in Rust, Python or C. The language that the corresponding service will be written in should determine
```
Comment (CatGarab) [inline docs/deep-dive/apis/device-api-guide.rst]: nit: comma after Python

File: docs/deep-dive/apis/device-api-guide.rst [hunk 6]
```diff
@@ -130,147 +76,136 @@ While each API is highly device-specific, our goal is to keep the overall stylin
...
+.. code-block:: c
 
-In order to generate the documentation locally, navigate to the top level of your copy of the Kubos repo and run ``tools/gendocs.py``.
+    // radio_device.c
 
-This will generate the documentation HTML files in a new ``html`` folder, which you can then use to verify your new docs display as intended.
+    static int radio_bus_handle = 0;
 
-To verify your docs:
+Each API should implement it's own ``Status`` enum.
```
Comment (CatGarab) [inline docs/deep-dive/apis/device-api-guide.rst]: nit: it's -> its

**Pull request merged**